script/phase-1d/31-LsmkrSpkFarmingDeploy.s.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {Script} from "forge-std/Script.sol";
+import {ScriptTools} from "dss-test/ScriptTools.sol";
+
+import {Reader} from "../helpers/Reader.sol";
+import {StakingRewardsDeploy, StakingRewardsDeployParams} from "../dependencies/StakingRewardsDeploy.sol";
+import {
+    VestedRewardsDistributionDeploy,
+    VestedRewardsDistributionDeployParams
+} from "../dependencies/VestedRewardsDistributionDeploy.sol";
+
+contract Phase1d_LsmkrSpkFarmingDeployScript is Script {
+    ChainlogLike internal constant chainlog = ChainlogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+
+    string internal constant NAME = "phase-1d/lsmkr-spk-farming-deploy";
+
+    function run() external {
+        Reader reader = new Reader(ScriptTools.loadConfig());
+
+        address admin = chainlog.getAddress("MCD_PAUSE_PROXY");
+
+        address lsmkr = reader.envOrReadAddress("FOUNDRY_LSMKR", ".lsmkr");
+        address spk = reader.envOrReadAddress("FOUNDRY_SPK", ".spk");
+        address vest = reader.envOrReadAddress("FOUNDRY_VEST", ".vest");
+        address dist = reader.readAddressOptional(".dist");
+        address rewards = reader.readAddressOptional(".rewards");
+
+        vm.startBroadcast();
+
+        if (rewards == address(0)) {
+            rewards = StakingRewardsDeploy.deploy(
+                StakingRewardsDeployParams({owner: admin, stakingToken: lsmkr, rewardsToken: spk})
+            );
+        }
+
+        if (dist == address(0)) {
+            dist = VestedRewardsDistributionDeploy.deploy(
+                VestedRewardsDistributionDeployParams({
+                    deployer: msg.sender,
```

Comment:
Foundry can have different msg.sender depending on how the script is run (option --sender) and wallet used to run. If the command is run with more than one signer and no --sender flag it defaults to the foundry default sender `0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38`. This could cause the deny call to be incorrect and the deployer to retain access to the contract. More details here: https://github.com/foundry-rs/foundry/issues/7255#issuecomment-2223057564

Reply:
How would that work? If the sender is incorrect, it wouldn't be able to sign the tx would it?

Reply:
Hey, I investigated this a bit further and [this line](https://github.com/makerdao/endgame-toolkit/blob/d785557c0b3d28607fd15132cc2fbad988fa32c2/script/dependencies/VestedRewardsDistributionDeploy.sol#L32) would fail if `p.deployer` is incorrect.

The workaround would be to always use `msg.sender`, because the only way for the script to work would be if the sender signs the transaction.

However, I'm not sure if it's worth to make that change at this point, since those libs are depended upon many other scripts in this repo.